pkg/rbac/parser.go

```diff
@@ -68,20 +126,36 @@ type Generator struct {
 func (Generator) RegisterMarkers(into *markers.Registry) error {
 	return into.Register(RuleDefinition)
 }
+
 func (g Generator) Generate(ctx *genall.GenerationContext) error {
-	var rules []rbacv1.PolicyRule
+	// the map key is the target of a Rule (Rule.target())
+	rules := make(map[string]*Rule)
```

Comment:
make the key a separate type -- it's clearer, and you can have `ruleKey { groups string; resources string; url string }`

Reply:
using ruleKey as the map key is a good suggestion. Fixed it!